src/unauthenticated_stream/mod.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+//! Unauthenticated stream cipher.
+//!
+//! # Warning
+//! This API does not provide any authentification! You should use the
+//! [aead](aead/index.html) module instead.
+//!
+//! C analog: `EVP_CIPHER`
+//!
+//! Go analog: [`crypto.cipher.Stream`]
+//!
+//! [`crypto.cipher.Stream`]: https://golang.org/pkg/crypto/cipher/#Stream
+
+use {chacha, error, init, polyfill};
+
+pub use self::chacha20::CHACHA20;
+
+/// A key for an unauthenticated stream cipher.
+///
+/// C analog: `EVP_CIPHER_CTX`
+///
+/// Go analog: [`crypto.cipher.Stream`]
+pub struct StreamingKey {
```

Comment:
Even though encryption and decryption are the same math for these ciphers, in *ring* we always have distinct public types for the encryption and decryption keys, and I'd like to continue that unless there is a reason not to. If there is a reason not to, what is it?